x/ccv/provider/keeper/relay.go

```diff
@@ -42,6 +41,21 @@ func (k Keeper) OnRecvVSCMaturedPacket(
 		panic(fmt.Errorf("VSCMaturedPacket received on unknown channel %s", packet.DestinationChannel))
 	}
 
+	// If no packets are in the per chain queue, immediately handle the vsc matured packet data
+	if k.GetPendingPacketDataSize(ctx, chainID) == 0 {
+		k.HandleVSCMaturedPacket(ctx, chainID, data)
+	} else {
```

Comment:
We could alternatively add a for-loop to the endblocker that iterates through all consumers, to handle VSC matured packet data that is queued before any slash packet data. This for-loop would happen before any of the global queue logic.

There's minimal benefit to this alternative, it's doing the same thing we're doing here, just on endblock

Reply:
I prefer the alternative you suggest because it groups the execution of vscids closer temporally. It saves having to reason about various module endblock orderings ect.

Reply:
superseded by https://github.com/cosmos/interchain-security/issues/573